src/herder/HerderImpl.cpp

```diff
@@ -914,13 +918,15 @@ HerderImpl::setInSyncAndTriggerNextLedger()
 
     // Trigger next ledger, without requiring Herder to properly track SCP
     auto lcl = mLedgerManager.getLastClosedLedgerNum();
-    triggerNextLedger(lcl + 1, false);
+    triggerNextLedger(make_optional<uint32_t>(lcl + 1), nullopt<TimePoint>(),
+                      false);
 }
 
 // called to take a position during the next round
 // uses the state in LedgerManager to derive a starting position
 void
-HerderImpl::triggerNextLedger(uint32_t ledgerSeqToTrigger,
+HerderImpl::triggerNextLedger(optional<uint32_t> ledgerSeqToTrigger,
```

Comment:
It looks that from the conversation https://github.com/stellar/stellar-core/pull/2696#discussion_r483277469 you went with the "virtual time" option.

In this case, you don't actually need to (or want to) change `triggerNextLedger` that should just work "out of the box" as to provide full fidelity; in fact I don't even think that you need to make any modification to herder and just make sure that you setup the "initial state" properly:

* if a close time is specified by `manualclose`, you should just "jump time ahead" (see `setCurrentVirtualTime`)
* if a ledger seq num is specified (and it requires "jumping ahead" from a ledger point of view), this should be done outside of `triggerNextLedger` by directly adjusting lcl

Reply:
Yes, you're right, once I went to `VIRTUAL_TIME`, the seeming need to modify `triggerNextLedger()` melted away.  :)

The new implementation is  in `ApplicationImpl::manualClose()`, and it does use `setCurrentVirtualTime()` and explicit modification of `loadHeader().current()`.